src/module.rs

```diff
@@ -118,6 +125,43 @@ impl<'a> fmt::Display for Module<'a> {
     }
 }
 
```

Comment:
```suggestion

/// Many shells cannot deal with raw unprintable characters (like ANSI escape sequences) and 
/// miscompute the cursor position as a result, leading to strange visual bugs. Here, we wrap these
/// characters in shell-specific escape codes to indicate to the shell that they are zero-length.
```

You can update the comment with something that makes more sense to you, but let's leave something for future code spelunkers to know why we did this ^^